node/src/actors/json_rpc/json_rpc_methods.rs

```diff
@@ -588,15 +591,44 @@ pub fn get_block(hash: Result<(Hash,), jsonrpc_core::Error>) -> JsonRpcResultAsy
                         .expect("The result of getBlock should be an object")
                         .insert("txns_hashes".to_string(), txns_hashes);
 
-                    futures::finished(value)
+                    // Check if this block is confirmed by a majority of superblock votes
+                    let chain_manager = ChainManager::from_registry();
+                    let fut = chain_manager
+                        .send(IsConfirmedBlock {
+                            block_hash,
+                            block_epoch,
+                        })
+                        .then(|res| match res {
+                            Ok(Ok(confirmed)) => {
+                                if confirmed {
+                                    // Append {"confirmed":true} to the result
+                                    value
+                                        .as_object_mut()
+                                        .expect("The result of getBlock should be an object")
+                                        .insert("confirmed".to_string(), Value::Bool(true));
```

Comment:
Is there a way we return a Value::Bool(false)?

Reply:
I mean, if there is any case for that. I dont see it and seems we should right?

Reply:
I assumed that if the "confirmed" key does not exist it is the same as setting it to false

Reply:
Ok, a bit confusing  but I understand it is to save some bytes in some cases. Although given the majority of blocks will have the true field maybe it should be the other way around haha

Reply:
lets leave it like that though, it might be more confusing otherwise

Reply:
I agree with @girazoki :smile:

Reply:
Ok, changed. Now all blocks should have a "confirmed" field.